pages/admin/gall.tsx

```diff
@@ -146,7 +146,7 @@ const Gall = ({
             cells: fields.cells[0].cells,
             color: fields.color[0].color,
             commonnames: fields.commonnames,
-            detachable: fields.detachable,
+            detachable: fields.detachable ? 1 : 0,
```

Comment:
The type of FormFields.detachable is string so I do not think that this logic works as we need it to. The real problem here is that handling of detachable is super hacky. I started with a boolean but at some point it ended up that we actually need 3 states, true/false/none. Currently it uses a string to track the values. I am thinking that it would make a lot of sense to clean this up by introducing a concrete type in [libs/api/apitypes.ts](https://github.com/jeffdc/gallformers/blob/main/libs/api/apitypes.ts) like this:
``` typescript
// using empty string rather than null/undefined to avoid having to deal with null/undefined later though it may not be a big deal to make undefined. I did not look into much.
type Detachable = true | false | ''; 
```
Then we have to make sure to map the DB value to this when we extract it. That happens in [/libs/db/gall.ts](https://github.com/jeffdc/gallformers/blob/0d51fa7bc0891049323262952dd4da80ee83a60a/libs/db/gall.ts#L101). We could then get rid of the [Option<number> in GallApi](https://github.com/jeffdc/gallformers/blob/0d51fa7bc0891049323262952dd4da80ee83a60a/libs/api/apitypes.ts#L274) and change it to Detachable.

Reply:
Ahh, yes, I think I better understand what's going on now. I talked with Adam about this field and how it's modeled. It seems like we may actually want to update it to go beyond a nullable bool. Maybe an option like you have with more elements (eg "both" or something). He's apparently going to explain in a comment in the issue.

Reply:
I see that. Seems like it will be a straight forward change that will build on the idea above. Once we get a solid write-up in the issue we can figure out how to tackle it.